Buttplug.Server.Managers.ErosTek/ETSerialManager.cs

```diff
@@ -0,0 +1,154 @@
+using System.Threading;
+using Buttplug.Core;
+using static Buttplug.Server.Managers.ETSerialManager.ET312Protocol;
+
+namespace Buttplug.Server.Managers.ETSerialManager
+{
+    public class ETSerialManager : DeviceSubtypeManager
+    {
+        private bool _isScanning = false;
+        private Thread _scanThread;
+        private string[] _searchComPorts;
+
+        public ETSerialManager(IButtplugLogManager aLogManager)
+            : base(aLogManager)
+        {
+            BpLogger.Info("Loading ErosTek Serial Port Manager");
+        }
+
+        public override void StartScanning()
+        {
+            BpLogger.Info("Starting Scanning Serial Ports for ErosTek Devices");
+            _scanThread = new Thread(() => ScanSerialPorts(_searchComPorts));
+            _scanThread.Start();
+        }
+
+        public override void StopScanning()
+        {
+            BpLogger.Info("Stopping Scanning Serial Ports for ErosTek Devices");
+            _isScanning = false;
+        }
+
+        public override bool IsScanning()
+        {
+            return _isScanning;
+        }
+
+        private void ScanSerialPorts(string[] selectedComPorts)
+        {
+            string[] comPortsToScan;
+            _isScanning = true;
+
+            while (_isScanning)
+            {
+                // Enumerate Ports
+                if (selectedComPorts == null || selectedComPorts.Length == 0)
+                {
+                    comPortsToScan = SerialPort.GetPortNames();
+                }
+                else
+                {
+                    comPortsToScan = selectedComPorts;
+                }
+
+                // try to detect devices on all port
+                foreach (var port in comPortsToScan)
+                {
+                    BpLogger.Info("Scanning " + port);
+
+                    SerialPort serialPort = new SerialPort(port)
+                    {
+                        ReadTimeout = 200,
+                        WriteTimeout = 200,
+                        BaudRate = 19200,
+                        Parity = Parity.None,
+                        StopBits = StopBits.One,
+                        DataBits = 8,
+                        Handshake = Handshake.None,
+                    };
+
+                    try
+                    {
+                        serialPort.Open();
+                    }
+                    catch (Exception ex)
+                    {
+                        if (ex is UnauthorizedAccessException
+                            || ex is ArgumentOutOfRangeException
+                            || ex is System.IO.IOException)
+                        {
+                            // This port is inaccessible.
+                            // Possibly because a device detected earlier is already using it,
+                            // or because our required parameters are not supported
+                            continue;
+                        }
+
+                        throw ex;
```

Comment:
Instead of `throw ex`, just use `throw` otherwise the stacktrace will start from this line onwards treating the exception as new instead of where the exception originally occurred.